src/config/index.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable */
+import axios from 'axios';
+import { SERVICE_CONFIGURATION_SERVICE_HTTP } from '../env';
+import { sleep } from './utils';
+import { ApplicationError } from '../errors';
+import { CHAIN_ID_MAINNET } from '../lib/constants';
+
+const serviceName = 'volumetracker';
+
+export const CORS_ALLOWED_HEADERS = 'Accept, Content-Type, Origin';
+export const REQUEST_BODY_SIZE_LIMIT_BYTES = 256 * 1024; // 256KB
+export const CACHE_CONTROL_PREFLIGHT_REQUESTS_MAX_AGE_SECS = 24 * 60 * 60; // 24 hours
+
+const CONFIG_SERVICE_TIMEOUT = 5000;
+const CONFIG_SERVICE_RETRY_INTERVAL = 3000;
+
+export type NetworkMap<T> = { [network: number]: T };
+
+export type Config = {
+  network: number;
+  augustusAddress: string;
+  augustusV4Address: string;
+  pspAddress: string;
+  isStaking: boolean;
+  rewardDistributionAddress: string;
+  safetyModuleAddress: string;
+  privateHttpArchiveProvider: string;
+  coinGeckoPlatform: string;
+  multicallV2Address: string;
+  volumeTrackerInitTime: number;
+};
+
+type GlobalConfig = {
+  apiKeyCoingecko: string;
+  covalentV1ApiKey: string;
+  covalentV1HttpUrl: string;
+  apiPrefineryHttp: string;
+  apiKeyPrefinery: string;
+  apiAplcapiHttp: string;
+  apiKeyAplcapi: string;
+  databaseUrl: string;
+  apiKeyCaptcha: string;
+  apiKeySubmitAccount: string;
+};
+
+type ConfigResponse = {
+  networks: NetworkMap<Config>;
+  global: GlobalConfig;
+};
+
+class ConfigLoader {
+  public byNetwork: NetworkMap<Config> = {};
+
+  public enabledNetworks: number[] = [];
+
+  public global?: GlobalConfig;
+
+  public isLoaded: Promise<void>;
+
+  public hasStartedNotifier?: (value: void | PromiseLike<void>) => void;
+
+  constructor() {
+    this.isLoaded = new Promise(resolve => {
+      this.hasStartedNotifier = resolve;
+    });
+  }
+
```

Comment:
As project is open and meant for any one to run it and challenge our computation I think it's important to make this an easy road. 
Perhaps we could allow a local config only mode ? 
Could be done by constructing a default config object where keys would be pocked from env var.